### DISCUSSION TITLE
TiddlyWiki5 Encryption with NodeJS

### DISCUSSION THREAD

**Discussion by jgeromero** [TiddlyWiki/TiddlyWiki5#8371] [Q&A] [answered]
Hello! I have a single file TW5 with password encryption (Stanford JavaScript Crypto Library) looking to host it on a Webserver locally. Should I expect this encryption to work the same way with NodeJS?

**Comment by Jermolene** [comment, marked as answer]
Hi @jgeromero the Node.js configuration is used to build encrypted single file wikis, but it doesn't support encryption for data on disk.

The Node.js threat model is that the server is assumed to be secure, and we worry about data being tampered with in transit, which is where HTTPS (TLS) comes in.

If you want to use the Node.js configuration on a machine in a secure way then you would probably be best off running it within an encrypted disk volume.
